blocks/library/block/index.php

```diff
@@ -13,8 +13,17 @@
  * @return string Rendered HTML of the referenced block.
  */
 function gutenberg_render_block_core_reusable_block( $attributes ) {
-	$reusable_block = get_post( $attributes['ref'] );
-	if ( ! $reusable_block ) {
+	if ( empty( $attributes['ref'] ) ) {
+		return '';
```

Comment:
`$attributes['ref']` will only be empty in two cases:

1. If there's a programming error elsewhere in our code (e.g. https://github.com/WordPress/gutenberg/pull/3879).
2. If the user has manually opened the Code Editor and typed in `<!-- wp:block /-->`.

With this in mind—should we be failing silently? I worry that by failing silently we will make it harder for us to notice bugs like https://github.com/WordPress/gutenberg/pull/3879 in the future.

Reply:
Great point. My preference would be to return a WP_Error in this unexpected scenario, but the current implementation expects a render callback like this one to return a string. So in my view an empty string is better than the other remaining alternative, throwing an exception.